devito/ir/iet/nodes.py

```diff
@@ -1492,6 +1495,9 @@ def body(self):
     def functions(self):
         return tuple(self.fmapper)
 
+    def __repr__(self):
+        funcs = self.halo_scheme.__reprfuncs__()
```

Comment:
again, it's a useless complication. Just return something along the lines of `HaloSpot(f,g)` and if the user really cares about the underlying `halo_scheme`, they can always do `hs.halo_scheme` and this way get the textual information they are after

so, IOW, drop `__reprfuncs__`